### DISCUSSION TITLE
Building for TV Apps, Amazon IAP not initialising connection?

### DISCUSSION THREAD

**Discussion by nnomadic** [hyochan/react-native-iap#1585] [Q&A]
So I've implemented all appropriate steps to get IAP set up for Amazon, but despite my best attempts, I always get `false` returned from `initConnection()`. I'm certain my config is correct as it initialises correctly and I'm able to go through the purchasing flow on Apple tvOS and Android TV, but when Live App Testing on an Amazon Fire TV stick, it simply does not connect. This also occurs on a local signed development build.

After digging around for quite a while, I'm assuming that the build variant used when building the APK is defaulting to the `playRelease` build variant for the module and not the `amazonRelease`. When I change this manually I get this error:

![Screenshot 2021-12-03 at 16 59 49](https://user-images.githubusercontent.com/15351439/144642467-3801c0ff-9ee6-4ff3-ae7c-c2719c23c5b4.png)

Is it even possible to select a build variant for a module manually? Am I thinking along the right lines here?  

If there's any more information I can provide for this please let me know.

**Comment by hyochan** [comment]
Would like to handle this in https://github.com/hyochan/expo-iap/discussions/239
